src/main/java/net/openhft/chronicle/threads/Threads.java

```diff
@@ -163,12 +163,9 @@ public static void interrupt(ExecutorService service) {
 
     static void forEachThread(ExecutorService service, Consumer<Thread> consumer) {
         try {
-            final Set workers;
-            if (!(service instanceof ThreadPoolExecutor)) {
-                workers = Jvm.getValue(resolveDelegatedExecutorServices(service), "workers");
-            } else {
-                workers = Jvm.getValue(service, "workers");
-            }
+            if (!(service instanceof ThreadPoolExecutor))
+                service = resolveDelegatedExecutorServices(service);
```

Comment:
this way when we try and get "mainLock" below we get it from the delegated service